### PR TITLE
Bloodsuckers - Frenzy & Clans

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This was made due to Swain's comments saying he'd like something like this by using Vampire clan. I re-used said vampire clan to deal with Sol, blocking him from using it, so I instead decided to make this as a proper alternative to use it Swain ever returns. 

EDIT: I gave up on the AI, wasn't good and isn't a good idea and forcing a Bloodsucker to just not be able to play isn't a really good idea.

## Changelog
:cl:
add: Bloodsuckers now enter Frenzy mode when they lose too much blood, where they lose all ability to speak, hear, interact with things and use their powers, until they seek Blood.
add: Bloodsuckers can now join a Clan at Rank 3, with each Clan giving good and bad effects.
refactor: Buying a new power now uses TGUI.
fix: Masquerade no longer lets Bloodsuckers automatically regenerate Blood.
fix: Persuasion racks now allow different types of tools to be used, and welders now harm their eyes.
fix: Bloodsucker's FinalDeath objective now takes into account whether you're in FinalDeath.
fix: Brawn now breaks lockers open from the outside.
fix: Bloodsuckers now properly regenerate any lost organs, and anything lost through Ahealing, by sleeping through Torpor.
fix: Fortitude can now be used in Torpor.
balance: Monster hunters now only get Hunter-Fu
balance: Immortal Haste now instantly grabs their target.
/:cl:

### CODE REVIEW

File: fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm [hunk 1]
```diff
@@ -344,8 +345,24 @@
...
+			)
+		switch(chosen_clan)
+			if(CLAN_BRUJAH)
+				my_clan = CLAN_BRUJAH
+			if(CLAN_NOSFERATU)
+				my_clan = CLAN_NOSFERATU
+			if(CLAN_TREMERE)
+				my_clan = CLAN_TREMERE
+			if(CLAN_VENTRUE)
+				my_clan = CLAN_VENTRUE
```
Comment (EgorDinamit) [inline fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm:364]: Can't you just:
```suggestion
		my_clan = pick(
			CLAN_BRUJAH,
			CLAN_NOSFERATU,
			CLAN_TREMERE,
			CLAN_VENTRUE,
			)
```
Unless you are planning to do epic stuff for each clan in here.

Comment (JohnFulpWillard) [reply fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm:364]: I originally did but moved it to the assignrandombane proc itself

File: fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_flaws.dm [hunk 2]
```diff
@@ -67,5 +67,24 @@
...
+					powers -= power
+					power.Remove(owner.current)
+			to_chat(owner, "<span class='announce'>You have Ranked up enough to learn: You are part of the Nosferatu Clan!<br> \
+				* As part of the Nosferatu Clan, you are less interested in disguising yourself within the crew, as such you do not know how to use the Masquerade ability.</span>")
+		if(my_clan == CLAN_TREMERE)
+			to_chat(owner, "<span class='announce'>You have Ranked up enough to learn: You are part of the Tremere Clan!<br> \
+				* As part of the Tremere Clan, you are weak to Anti-magic, and will catch fire if you enter the Chapel!</span>")
+		if(my_clan == CLAN_VENTRUE)
+			to_chat(owner, "<span class='announce'>You have Ranked up enough to learn: You are part of the Ventrue Clan!<br> \
+				* As part of the Ventrue Clan, you are extremely snobby with your meals, and refuse to drink blood from people without a Mind.</span>")
```
Comment (SgtHunk) [inline fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_flaws.dm:87]: Friendly bloodsucker BTFO

Comment (JohnFulpWillard) [reply fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_flaws.dm:87]: wins

**Comment by JohnFulpWillard** [comment]
wtf github wont remove merge conflict ???

**Comment by JohnFulpWillard** [comment]
image
REMOVE MY LABEL YOU STUPID FUCKING BOT

**Pull request merged**